src/Microsoft.ML.Data/Evaluators/MulticlassClassificationEvaluator.cs

```diff
@@ -461,11 +477,21 @@ public override void ProcessRow()
                     // Penalize logloss if the label was not seen during training
                     logloss = -Math.Log(Epsilon);
                     _numUnknownClassInstances++;
+                    wasKnownLabel = false;
                 }
 
-                UnweightedCounters.Update(_indicesArr, logloss, intLabel, 1);
+                // Get the probability that the CORRECT label has: (best case is that it's the highest probability):
+                var correctProba = !wasKnownLabel ? 0 : _scoresArr[intLabel];
+
+                // Find the rank of the *correct* label (in Scores[]). If 0 => Good, correct. And the lower the better.
```

Comment:
Nit: Instead of `Scores[]` it should be `_scoresArr[]`. Also I think it might be clearer if the second sentence read something like "If rank = 0, it means the correct label was predicted. If rank = 1, it meants the correct label was predicted as the 2nd highest prediction. And the lower the rank the better the prediction was". Or something along those lines.